src/Api/AdminConsole/Authorization/OrganizationClaimsExtensions.cs

```diff
@@ -0,0 +1,123 @@
+﻿#nullable enable
+
+using System.Security.Claims;
+using Bit.Core.Context;
+using Bit.Core.Enums;
+using Bit.Core.Identity;
+using Bit.Core.Models.Data;
+
+namespace Bit.Api.AdminConsole.Authorization;
+
+public static class OrganizationClaimsExtensions
+{
+    /// <summary>
+    /// A delegate that returns true if the user has the specified claim type for an organization, false otherwise.
+    /// </summary>
+    private delegate bool HasClaim(string claimType);
```

Comment:
I wouldn't do a custom delegate and instead do `Func<string, bool>`. I don't think we technically follow the framework design guidelines here but I try to. 

https://learn.microsoft.com/en-us/dotnet/standard/design-guidelines/events-and-callbacks

> DO use the new `Func<...>`, `Action<...>`, or `Expression<...>` types instead of custom delegates, when defining APIs with callbacks.

Reply:
I've seen this before and don't quite understand it as a general rule. Sometimes it's nice to give things names rather than infer from context or xmldoc. Not a big deal either way though so I've implemented your feedback.

Reply:
Thanks for the explanation!